fvm/src/kernel/mod.rs

```diff
@@ -115,6 +123,13 @@ pub trait Kernel:
     ) -> Result<CallResult>;
 }
 
+pub trait SyscallHandler {
+    fn bind_syscalls(
+        &self,
+        linker: &mut Linker<InvocationData<impl Kernel + 'static>>,
```

Comment:
Ok, so, I _think_ the solution here may be:

```rust
pub trait SyscallHandler<K: Kernel> {
    fn bind_syscalls(
        &self,
        linker: &mut Linker<InvocationData<K>>,
    ) -> anyhow::Result<()>;
}
```

You'd implement:

```rust
impl<K: Kernel> SyscallHandler<K> for BaseKernel {
    fn bind_syscalls(
        &self,
        linker: &mut Linker<InvocationData<K>>,
    ) -> anyhow::Result<()> { .. }
}

impl SyscallHandler<FilecoinKernel> for FilecoinKernel {
    fn bind_syscalls(
        &self,
        linker: &mut Linker<InvocationData<FilecoinKernel>>,
    ) -> anyhow::Result<()> { .. }
}
```

(maybe...)

Reply:
I think in order to get this to work, I must first get the ambassador stuff to work. Unfortunately, it does not seem to support traits that extend other traits (like Kernel) so I need to consolidate everything into a single Kernel trait; will see if that works...

Reply:
You'd need to use ambassador on each trait.

Reply:
yebb that was the issue, still working on getting it to work but getting closer I hope